go/vt/vttablet/tabletserver/vstreamer/planbuilder.go

```diff
@@ -601,10 +610,6 @@ func (plan *Plan) analyzeWhere(vschema *localVSchema, where *sqlparser.Where) er
 			if !ok {
 				return fmt.Errorf("unexpected: %v", sqlparser.String(expr))
 			}
-			// StrVal is varbinary, we do not support varchar since we would have to implement all collation types
```

Comment:
This support was actually added here: https://github.com/vitessio/vitess/pull/9549

Reply:
And it's tested in the new e2e test.